src/vcpkg/build.cpp

```diff
@@ -849,6 +851,189 @@ namespace vcpkg::Build
         }
     }
 
+    struct Sha512Scanner
```

Comment:
I do _not_ love parsing CMake output like this; I'd prefer to have an obvious, guid surrounded output that tells you exactly how to fix the sha512, something like:

```
{start-guid}
sha512-mismatch:
expected=<sha512>
found=<sha512>
{end-guid}
```

Reply:
Yeah same, but I have not found a better answer. The output is always also visible to the user.  

Hm ok one can filter out the output if there is the `{start-guid} ... {end-guid}` sequence and print a new error message when the replacing was not successful. I will try this.

Reply:
It now uses a guid based approach